nncf/sparsity/base_algo.py

```diff
@@ -161,3 +161,6 @@ def add_algo_specific_stats(self, stats):
 
     def compression_level(self) -> CompressionLevel:
         return CompressionLevel.FULL
+
+    def do_specific_setting_after_each_epoch(self):
+        pass
```

Comment:
Perhaps I misunderstand the purpose of this function, 'm not sure that we need it in general. Even now, this function is named too broadly, and the way it is used does not immediately suggest that it is indeed called "after each epoch" - it is also called in init, and in `_set_sparsity_level`. Maybe it should be more fittingly called `_run_post_sparsity_level_update_actions`? Does this reflect the way you wanted this function to be used? Could it be possibly merged with an algo-specific `_set_sparsity_level` then?

Reply:
Done.